src/definitions.jl

```diff
@@ -59,7 +59,7 @@ _to1(::Tuple, x) = copy1(eltype(x), x)
 for f in (:fft, :bfft, :ifft, :fft!, :bfft!, :ifft!, :rfft)
     pf = Symbol("plan_", f)
     @eval begin
-        $f(x::AbstractArray) = $f(x, 1:ndims(x))
+        $f(x::AbstractArray) = $f(x, ntuple(identity, ndims(x)))
```

Comment:
Can we just use
```suggestion
        $f(x::AbstractArray) = $f(x, Base.OneTo(ndims(x)))
```
? Maybe that already fixes constant propagation? It would be less breaking (I think) and should also be more efficient if `ndims(x)` is large.

Reply:
I don't think this necessarily fixes the issue without the compiler propagating `ndims(x)` as a constant.

Regarding large `ndims`, IIUC `Tuples` are performant till a length of 32, and it's uncommon to have arrays with that many dimensions. In any case, this could be an optimization for `ndims(x)<4`, which is perhaps the common case.

I fully understand and sympathize with your concern about breakages. I'll see if the direct dependants can be fixed first so that they accept arbitrary `regions`, before this is considered